// ===================================================================== \\
// ===================================================================== \\
// ========================  _____ ========  _ 	© ====================== \\
// ======================== | ___ \ ======= | | ======================== \\
// ======================== | |_/ / __ _  __| |	======================== \\
// ======================== | ___ \/ _` |/ _` |	======================== \\
// ======================== | |_/ / (_| | (_| |	======================== \\
// ======================== \____/ \__,_|\__,_|	======================== \\
// ===================================================================== \\
// ===================================================================== \\
// Developed by Gabriel dos Prazeres - [Bad] © ========================= \\
// All rights reserved. Contact: +55 91 99322-5657 ===================== \\
// ===================================================================== \\
// ===================================================================== \\
// Mapflag: NoRodex ==================================================== \\
// ===================================================================== \\
// Compatible with: Emulator rAthena - 02/10/2021 ====================== \\
// ===================================================================== \\
// Description: ======================================================== \\
// Blocks reading and writing of email by Rodex. ======================= \\
// ===================================================================== \\
// Version 1.0 ========================================================= \\
// ===================================================================== \\
// Discord: Bad#5743 =================================================== \\
// ===================================================================== \\

From a985dc4a4ee8b4c592381519e3356e815d397e5c Mon Sep 17 00:00:00 2001
From: Gabriel dos Prazeres <gabrieldosprazeres@hotmail.com.br>
Date: Mon, 4 Oct 2021 11:36:32 -0300
Subject: [PATCH] Finished mapflag NoRodex

---
 conf/msg_conf/map_msg.conf   | 4 ++++
 src/map/clif.cpp             | 8 ++++++++
 src/map/map.hpp              | 1 +
 src/map/script_constants.hpp | 1 +
 4 files changed, 14 insertions(+)

diff --git a/conf/msg_conf/map_msg.conf b/conf/msg_conf/map_msg.conf
index 4b10ebfba..42c9e4002 100644
--- a/conf/msg_conf/map_msg.conf
+++ b/conf/msg_conf/map_msg.conf
@@ -466,6 +466,10 @@
 
 451: Cash Shop is disabled on this map.
 
+// NoRodex messages [Bad]
+452: You cannot read email on this map.
+453: You cannot write email on this map.
+
 // Message System
 460: Please enter a valid language (usage: @langtype <language>).
 461: Language is now set to %s.
diff --git a/src/map/clif.cpp b/src/map/clif.cpp
index d812c12c1..468634bf5 100644
--- a/src/map/clif.cpp
+++ b/src/map/clif.cpp
@@ -15991,6 +15991,10 @@ void clif_parse_Mail_read(int fd, struct map_session_data *sd){
 		return;
 	if( mail_invalid_operation(sd) )
 		return;
+	if (map_getmapflag(sd->bl.m,MF_NORODEX)) { // [Bad]
+		clif_displaymessage(sd->fd, msg_txt(sd,452)); // You cannot read email on this map.
+		return;
+	}
 
 	clif_Mail_read(sd, mail_id);
 }
@@ -16017,6 +16021,10 @@ void clif_parse_Mail_beginwrite( int fd, struct map_session_data *sd ){
 		clif_send_Mail_beginwrite_ack(sd, name, false);
 		return;
 	}
+	if (map_getmapflag(sd->bl.m,MF_NORODEX)) { // [Bad]
+		clif_displaymessage(sd->fd, msg_txt(sd,453)); // You cannot write email on this map.
+		return;
+	}
 
 	mail_clear(sd);
 
diff --git a/src/map/map.hpp b/src/map/map.hpp
index 62ce03dac..7842f76bd 100644
--- a/src/map/map.hpp
+++ b/src/map/map.hpp
@@ -610,6 +610,7 @@ enum e_mapflag : int16 {
 	MF_PRIVATEAIRSHIP_DESTINATION,
 	MF_SKILL_DURATION,
 	MF_NOCASHSHOP,
+	MF_NORODEX, // [Bad]
 	MF_MAX
 };
 
diff --git a/src/map/script_constants.hpp b/src/map/script_constants.hpp
index 9e80a7ee4..81aef1ac0 100644
--- a/src/map/script_constants.hpp
+++ b/src/map/script_constants.hpp
@@ -479,6 +479,7 @@
 	export_constant(MF_PRIVATEAIRSHIP_DESTINATION);
 	export_constant(MF_SKILL_DURATION);
 	export_constant(MF_NOCASHSHOP);
+	export_constant(MF_NORODEX); // [Bad]
 
 	/* setcell types */
 	export_constant(CELL_WALKABLE);
-- 
2.33.0.windows.2

